### PR TITLE
Coverity reports dead code

### PR DESCRIPTION
**Pull request by zeertzjq**
Problem:  Coverity reports dead code.
Solution: Remove the dead code. Also fix a mistake and some comments.

### CODE REVIEW

File: src/ops.c [hunk 1]
```diff
@@ -2687,9 +2684,10 @@ do_addsub(
 
     curwin->w_cursor = *pos;
     ptr = ml_get(pos->lnum);
+    linelen = ml_get_len(pos->lnum);
     col = pos->col;
 
-    if (*ptr == NUL || col + !!save_coladd >= (int)STRLEN(ptr))
```
Comment (chrisbra) [inline src/ops.c:2692]: why is the `*ptr == NUL` condition removed?

Comment (zeertzjq) [reply src/ops.c:2692]: It's not needed because `col >= 0`, and `ml_get_len()` returns 0 if `*ptr == NUL`.

File: src/memline.c [hunk 2]
```diff
@@ -2686,7 +2686,7 @@ ml_get_len(linenr_T lnum)
     colnr_T
 ml_get_pos_len(pos_T *pos)
 {
-    return ml_get_buf_len(curbuf, curwin->w_cursor.lnum) - pos->col;
+    return ml_get_buf_len(curbuf, pos->lnum) - pos->col;
```
Comment (chrisbra) [inline src/memline.c:2689]: oh, I should have caught this when reviewing yesterday :/

File: src/ops.c (hunk 1 above)
Comment (chrisbra) [reply src/ops.c:2692]: ah, so when `*ptr == NUL` this means that the second condition will be also true: `col + !!save_coladd >= linelen` I see. thanks

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**